SofairOfficial/rust-repository-template
# feat(service): implement a tiny greeter service

## Description

This pull request (PR) implements a sample greeter service library and a "command-line interface" (CLI) application to execute the service.

This sample greeter service is mostly used to test the [CI/CD pipeline]() operations, including code style checking, testing, and so on.

## Related issues

This PR closes issue #15  

## How has this been tested?

Only unit tests, for now, are implemented, that can be executed as follow:

`$ cargo test`

To test Clippy you should proceed as follows:

`$ cargo clippy --all-targets --all-features -- -Dwarnings`

To check Rustfmt operations, please enter the following command:

`$ cargo fmt`

**Test configuration**:
* Hardware: MacOS Ventura 13.2.1
* Toolchain: Rust 1.67.0
* SDK: -

## Checklist:
